Observer/AfterCheckoutSubmitObserver.php

```diff
@@ -106,8 +117,30 @@ public function execute(Observer $observer)
         $this->paymentData->update($paymentData);
 
         if ($paymentData->getIntent() == 'Sale') {
+            $this->logger->debug('Saving sale transaction number to order grid!');
```

Comment:
Since `saveTransactionNumber` may not find a `Sale` transaction, perhaps it would be better to move the log statement to when the `transaction.number` is found and also include the transaction number in the log statement?